### PR TITLE
Advert Effects & Aim Help

### PR DESCRIPTION
**Pull request by telmiuse**
AimCursor.Scene & AdvertSecene.scene

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-1e6nwjw Spots Motion Effect](https://app.clickup.com/t/1e6nwjw)

**Comment by telmiuse** [comment]
gif

**Comment by telmiuse** [comment]
UFXKvg3Ljk (online-video-cutter com)

**Review by JaimeBea — DISMISSED** [review]

**Comment by JaimeBea** [comment]
Bugs:

- Laser keeps working while using ULT and Q

File: Penteract/Assets/Scripts/AdvertScrip.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+EXPOSE_MEMBERS(AdvertScrip) {
+        MEMBER(MemberType::FLOAT, speed),
+        MEMBER(MemberType::FLOAT, startTime),
+        MEMBER(MemberType::FLOAT, changeTime),
+};
+
+GENERATE_BODY_IMPL(AdvertScrip);
+
+void AdvertScrip::Start() {
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/AdvertScrip.cpp:13]: ScripT

File: Penteract/Assets/Scripts/AdvertScrip.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                if (offset.y < 0.5) {
+                    offset += float2(0, speed) * Time::GetDeltaTime();
+                    GetOwner().GetComponent<ComponentMeshRenderer>()->SetTextureOffset(offset);
+                }
+                else {
+                    offset.y = 0.5;
+                    isChanging = false;
+                }
+            }
+            if (!firstAdvert) {
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/AdvertScrip.cpp:34]: Why is this not an else statement?

**Review by Mefiso — COMMENTED** [review]
The scene AdvertSecene.scene doesn't seem to have the advertisements working correctly. Review if it also happens on your computer, pls.

File: Penteract/Assets/Scripts/CameraController.cpp [hunk 3]
```diff
@@ -42,12 +43,30 @@ void CameraController::Update() {
...
-
 	float3 desiredPosition = playerGlobalPos + float3(cameraOffsetX, cameraOffsetY, cameraOffsetZ);
-	float3 smoothedPosition = desiredPosition;
 
+	float3 smoothedPosition = desiredPosition;
+		
 	if (useSmoothCamera) {
-		smoothedPosition = float3::Lerp(transform->GetGlobalPosition(), desiredPosition, smoothCameraSpeed * Time::GetDeltaTime());
+		if (playerController->playerFang.isAiming()) {
+			float2 mousePosition = Input::GetMousePositionNormalized();
```
Comment (devildrake) [inline Penteract/Assets/Scripts/CameraController.cpp:52]: Does this not take into account gamepad at all, is gamepad discarded?

Comment (telmiuse) [reply Penteract/Assets/Scripts/CameraController.cpp:52]: i dont have the xbox controller to addet yet, ill put this feature in TODO

File: Penteract/Assets/Scripts/CameraController.cpp [hunk 4]
```diff
@@ -42,12 +44,30 @@ void CameraController::Update() {
...
+
+			if (mousePosition.x > 0.75) { aimingPositionX = aimingDistance; }
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/CameraController.cpp:55]: It would be nice to have this threshold value exposed, but let's do that in further PRs

**Review by magalenyo — APPROVED** [review]

**Review by Mefiso — APPROVED** [review]

**Review by JaimeBea — APPROVED** [review]

**Pull request merged**